CMakeLists.txt

```diff
@@ -268,90 +264,9 @@ else()
   set(OPENRAVE_LIBRARY_SUFFIX "${OPENRAVE_SOVERSION}" CACHE STRING "Suffix to append to library names")
 endif()
 
-# only look for installed boost if not an old version of msvc
-if(NOT MSVC71 AND NOT MSVC80)
-  if( NOT $ENV{BOOST_INCLUDEDIR} STREQUAL "" )
-    set(Boost_INCLUDE_DIR $ENV{BOOST_INCLUDEDIR})
-  endif()
-  if( NOT $ENV{BOOST_LIBRARYDIR} STREQUAL "" )
-    set(Boost_LIBRARY_DIRS $ENV{BOOST_LIBRARYDIR})
-  endif()
-  set(Boost_ADDITIONAL_VERSIONS "1.46.1" "1.45" "1.44" "1.43" "1.42" "1.41" "1.40" "1.39" "1.38" "1.37.0" "1.37" "1.35.0" "1.34.1" "1.34.0" "1.34" "1.33.1" "1.33.0" "1.33" ${Boost_ADDITIONAL_VERSIONS})
-  if(OPT_PYTHON AND NOT USE_PYBIND11_PYTHON_BINDINGS)
-    find_package(Boost COMPONENTS regex filesystem system python thread iostreams date_time)
-  else()
-    find_package(Boost COMPONENTS regex filesystem system thread iostreams date_time)
-  endif()
-else()
-  # have to rely on local boost and a hack
-  add_definitions(-DTEMP_SPIRIT_HACK)
-endif()
-
-if( Boost_FOUND )
-  include_directories(${Boost_INCLUDE_DIRS})
-  set(OPENRAVE_LINK_DIRS ${OPENRAVE_LINK_DIRS} ${Boost_LIBRARY_DIRS})
-elseif(Boost_VERSION AND NOT "${Boost_VERSION}" STREQUAL "0")
-  include_directories(${Boost_INCLUDE_DIRS})
-  set(OPENRAVE_LINK_DIRS ${OPENRAVE_LINK_DIRS} ${Boost_LIBRARY_DIRS})
-elseif( MSVC )

```

Comment:
not sure about msvc, but maybe okay, we don't support msvc for a while..

Reply:
worrying msvc caused https://github.com/rdiankov/openrave/pull/1227

Reply:
msvc has evolved over the years and so has cmake, so i don't mind dropping msvc support for now since the current way of linking might not even work anymore. we can figure out how to re-add it back when we're actually testing on the latest versions.

Reply:
Linking libraries with CMake on Windows and Linux all work the same way, and Boost has had proper CMake support for almost a decade now. It is not necessary to write janky workarounds anymore.